packages/skeleton/src/lib/utilities/Modal/Modal.svelte

```diff
@@ -131,6 +134,19 @@
 		currentComponent = typeof modals[0].component === 'string' ? components[modals[0].component] : modals[0].component;
 	});
 
+	function onModalHeightChange(modalHeight: number | null) {
+		// modal is closed
+		if (!modalHeight) return;
+
+		if (modalHeight > windowHeight) {
+			cBackdropOverflow = 'overflow-y-auto';
+		} else {
+			cBackdropOverflow = 'overflow-y-hidden';
+		}
+	}
+	// first child of the modal is the content.
+	$: onModalHeightChange((modalElement?.firstChild as HTMLElement)?.clientHeight);
```

Comment:
I'm not sure `firstChild` is reliable here. While we suggest users created their custom modal components with a single wrapping element, there's no guarantee they will. Any reason not to just verify the size of the modal element itself? It should always be present and always wrap the entirety of the content.

If you're concerned about padding and borders - then I'd recommend the use of `offsetHeight`:
https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/offsetHeight

Reply:
@endigo9740 
while you are correct about the `firstChild` not being reliable, the modal element itself is also not reliable, when using custom modals the modal-element has the height of (0) always. so I have changed the code to check both the parent and the child before deciding which one represents the size. this should work with every situation.